sam/pileup.go

```diff
@@ -437,3 +438,200 @@ func (p *Pile) String() string {
 	return fmt.Sprintf("RefIdx: %d\tPos: %d\tCountF: %v\tCountR: %v\tInsCountF: %v\tInsCountR: %v\tDelCountF: %v\tDelCountR: %v\tNext: %p\tPrev: %p",
 		p.RefIdx, p.Pos, p.CountF, p.CountR, p.InsCountF, p.InsCountR, p.DelCountF, p.DelCountR, p.next, p.prev)
 }
+
```

Comment:
For the sake of my own future sanity as I am debugging, could I ask you to move the consensus stuff to a new file (maybe pileupConsensus.go?). pileup.go is already a pretty crowded file.